packages/jaeger-ui/src/types/metrics.tsx

```diff
@@ -23,14 +23,16 @@ export type AvailableOpsMetrics =
   | 'service_operation_latencies'
   | 'service_operation_error_rate';
 
+export type spanKinds = 'unspecified' | 'internal' | 'server' | 'client' | 'producer' | 'consumer';
+
 export type MetricsAPIQueryParams = {
   quantile: number;
   groupByOperation?: boolean;
   endTs?: number;
   lookback?: number;
   step?: number;
   ratePer?: number;
-  spanKind?: 'unspecified' | 'internal' | 'server' | 'client' | 'producer' | 'consumer';
+  spanKind?: spanKinds;
```

Comment:
lots of optional fields here - are they really optional? Even if they are from the server perspective, wouldn't UI code always populate them?

Reply:
Hmmm.. yeah, not sure why it was done that way initially; removed optionality in: [ccd6cc7](https://github.com/jaegertracing/jaeger-ui/pull/2012/commits/ccd6cc76fc4f4d34178978ce7b3a4f03aaab0182).